src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,354 @@
+                self.assertLessEqual( grants_of_cbsd[0]['requestedOperationParam']\
+                                 ['operationFrequencyRange']['highFrequency'], 3700000000)
+                self.assertEqual( grants_of_cbsd[0]['requestedOperationParam']\
+                                 ['operationFrequencyRange']['highFrequency'] % 5000000, 0)     
+            self.assertDictEqual( grants_of_cbsd[0]['operationParam'], \
+                                  grant_request[index]['operationParam'])
+            self.assertEqual(grants_of_cbsd[0]['channelType'], grant_response[index]['channelType'])
+            self.assertEqual( grants_of_cbsd[0]['grantExpireTime'], grant_response[index]['grantExpireTime'])
+            self.assertFalse(grants_of_cbsd[0]['terminated'])
+    
+    def generate_FAD_1_default_config(self, filename):
+        """Generates the WinnForum configuration for FAD.1"""
+        # Load device info
+        device_a = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        device_b = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_b.json')))
+        device_c = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_c.json')))
+        # Device #2 is Category B
+        # pre-loaded.
+        self.assertEqual(device_b['cbsdCategory'], 'B')
+        conditionals_b = {
+            'cbsdCategory': device_b['cbsdCategory'],
+            'fccId': device_b['fccId'],
+            'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+            'airInterface': device_b['airInterface'],
+            'installationParam': device_b['installationParam']
+        }
+        conditionals = {'registrationData': [conditionals_b]}
+        del device_b['cbsdCategory']
+        del device_b['airInterface']
+        del device_b['installationParam']
+        devices = [device_a, device_b, device_c]
+        # Grants
+        grants = []
+        for index in range(len(devices)):
+            grants.append(json.load(
+             open(os.path.join('testcases', 'testdata', 'grant_0.json'))))
+        # PPAs and PALs
+        ppas = []
+        pals = []
+        pal_low_frequency = 3550000000.0
+        pal_high_frequency = 3560000000.0
+        pal_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        ppa_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+        ppa, pal = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        ppas.append(ppa)
+        pals.append(pal)
```

Comment:
I think you want to use `extend` here, otherwise you will have a list of lists below. Note that `pal` here is actually an array based on the output format of `makePpaAndPalRecordsConsistent`.

Reply:
I fixed that, thank you.